`ErrorObject` attributes are not visible to mypy

### Describe the bug

Since type annotations have been exported, many errors are now being reported for missing attributes on `ErrorObject` instances.

### To Reproduce

in `t.py`:

```python
import stripe

def f() -> None:
    try:
        stripe.PaymentIntent.create()
    except stripe.error.InvalidRequestError as e:
        error = e.construct_error_object()
        assert error is not None  # not relevant to this issue
        print(error.code)
```

```console
$ mypy t.py
t.py:9: error: "ErrorObject" has no attribute "code"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

### Expected behavior

No type checking errors.

### Code snippets

_No response_

### OS

macOS

### Language version

observed on 3.10.11 and 3.11.5

### Library version

7.1.0

### API version

N/A

### Additional context

_No response_

--- a/stripe/api_resources/error_object.py
+++ b/stripe/api_resources/error_object.py
@@ -9,6 +9,7 @@ if TYPE_CHECKING:
     from stripe.api_resources.source import Source
     from stripe.api_resources.payment_method import PaymentMethod
 
+
 class ErrorObject(StripeObject):
     charge: Optional[str]
     code: int

--- a/stripe/api_resources/error_object.py
+++ b/stripe/api_resources/error_object.py
@@ -2,10 +2,9 @@ from typing import Optional
 from stripe.api_resources.payment_intent import PaymentIntent
 from stripe.api_resources.setup_intent import SetupIntent
 from stripe.api_resources.source import Source
+from stripe.api_resources.payment_method import PaymentMethod
 from stripe.util import merge_dicts
 from stripe.stripe_object import StripeObject
-
-from api_resources.payment_method import PaymentMethod
 
 
 class ErrorObject(StripeObject):

--- a/stripe/api_resources/error_object.py
+++ b/stripe/api_resources/error_object.py
@@ -1,8 +1,26 @@
+from typing import Optional
+from stripe.api_resources.payment_intent import PaymentIntent
+from stripe.api_resources.setup_intent import SetupIntent
+from stripe.api_resources.source import Source
 from stripe.util import merge_dicts
 from stripe.stripe_object import StripeObject
 
+from api_resources.payment_method import PaymentMethod
+
 
 class ErrorObject(StripeObject):
+    charge: Optional[str]
+    code: int
+    decline_code: Optional[str]
+    doc_url: Optional[str]
+    message: Optional[str]
+    param: Optional[str]
+    payment_intent: Optional[PaymentIntent]
+    payment_method: Optional[PaymentMethod]
+    setup_intent: Optional[SetupIntent]
+    source: Optional[Source]
+    type: str
+
     def refresh_from(
         self,
         values,

--- a/stripe/api_resources/error_object.py
+++ b/stripe/api_resources/error_object.py
@@ -1,11 +1,13 @@
 from typing import Optional
-from stripe.api_resources.payment_intent import PaymentIntent
-from stripe.api_resources.setup_intent import SetupIntent
-from stripe.api_resources.source import Source
-from stripe.api_resources.payment_method import PaymentMethod
+from typing_extensions import TYPE_CHECKING
 from stripe.util import merge_dicts
 from stripe.stripe_object import StripeObject
 
+if TYPE_CHECKING:
+    from stripe.api_resources.payment_intent import PaymentIntent
+    from stripe.api_resources.setup_intent import SetupIntent
+    from stripe.api_resources.source import Source
+    from stripe.api_resources.payment_method import PaymentMethod
 
 class ErrorObject(StripeObject):
     charge: Optional[str]
@@ -14,10 +16,10 @@ class ErrorObject(StripeObject):
     doc_url: Optional[str]
     message: Optional[str]
     param: Optional[str]
-    payment_intent: Optional[PaymentIntent]
-    payment_method: Optional[PaymentMethod]
-    setup_intent: Optional[SetupIntent]
-    source: Optional[Source]
+    payment_intent: Optional["PaymentIntent"]
+    payment_method: Optional["PaymentMethod"]
+    setup_intent: Optional["SetupIntent"]
+    source: Optional["Source"]
     type: str
 
     def refresh_from(

